python/semantic_kernel/functions/kernel_function_from_prompt.py

```diff
@@ -0,0 +1,223 @@
+        """
+        if not prompt and not prompt_template_config and not prompt_template:
+            raise KernelFunctionInitializationException(
+                "The prompt cannot be empty, must be supplied directly, \
+through prompt_template_config or in the prompt_template."
+            )
+
+        if not prompt_template:
+            if not prompt_template_config:
+                # prompt must be there if prompt_template and prompt_template_config is not supplied
+                prompt_template_config = PromptTemplateConfig(
+                    name=function_name,
+                    description=description,
+                    template=prompt,
+                    template_format=template_format,
+                )
+            prompt_template = KernelPromptTemplate(prompt_template_config=prompt_template_config)
+
+        try:
+            metadata = KernelFunctionMetadata(
+                name=function_name,
+                plugin_name=plugin_name,
+                description=description,
+                parameters=prompt_template.prompt_template_config.get_kernel_parameter_metadata(),
+                is_prompt=True,
+                is_asynchronous=True,
+                return_parameter=PROMPT_RETURN_PARAM,
+            )
+        except ValidationError as exc:
+            # reraise the exception to clarify it comes from KernelFunctionFromPrompt init
+            raise exc
+        super().__init__(
+            metadata=metadata, prompt_template=prompt_template, prompt_execution_settings=prompt_execution_settings
+        )
+
+    @model_validator(mode="before")
+    @classmethod
+    def rewrite_execution_settings(
+        cls,
+        data: Dict[str, Any],
+    ) -> Dict[str, PromptExecutionSettings]:
+        """Rewrite execution settings to a dictionary.
+
+        If the prompt_execution_settings is not a dictionary, it is converted to a dictionary.
+        If it is not supplied, but prompt_template is, the prompt_template's execution settings are used.
+        """
+        prompt_execution_settings = data.get("prompt_execution_settings")
+        prompt_template = data.get("prompt_template")
+        if not prompt_execution_settings:
+            if prompt_template:
+                prompt_execution_settings = prompt_template.prompt_template_config.execution_settings
+                data["prompt_execution_settings"] = prompt_execution_settings
+            if not prompt_execution_settings:
+                return data
+        if isinstance(prompt_execution_settings, PromptExecutionSettings):
+            data["prompt_execution_settings"] = {
+                prompt_execution_settings.service_id or "default": prompt_execution_settings
+            }
+        if isinstance(prompt_execution_settings, list):
+            data["prompt_execution_settings"] = {s.service_id or "default": s for s in prompt_execution_settings}
+        return data
+
+    async def _invoke_internal(
```

Comment:
No need to call `gather_function_parameters()` as part of this method?

Reply:
no, the internal function uses kernel and arguments, and with those they are complete and good to go.